src/components/context/CurrentUserContext.js

```diff
@@ -17,8 +17,11 @@ const CurrentUserContextProvider = ({ children }) => {
     }
   }, [lang]);
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const value = useMemo(() => query, [query.data]);
```

Comment:
Does this really cache the query value? Isn't it passed by reference? So if `query.data` changes, the nested reference will be updated anyway, so I guess `query` is not immutable. Am I missing something?

Reply:
I needed it because it was infinitely looping for some reasons. It's not the best solution because it won't provide the most accurate isError/isLoading... of the value. But it's anyway not supposed to change much.